app/models/snack.rb

```diff
@@ -0,0 +1,9 @@
+class Snack < ApplicationRecord
+  validates_presence_of :name, :price
+  has_many :machine_snacks
+  has_many :machines, through: :machine_snacks
+
+  def locations
+    machines.pluck(:location)
+  end
```

Comment:
Oh interesting, I never thought to do this, but good use of a class method, and activerecord!